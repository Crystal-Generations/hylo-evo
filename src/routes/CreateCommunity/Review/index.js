import React, { Component } from 'react'
import '../CreateCommunity.scss'
import LeftSidebar from '../../Signup/LeftSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from '../ModalFooter'

const theme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}
export default class Review extends Component {
  setState = () => {

  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header='Everything looking good?'
        body='You can always come back and change your details at any time'
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 4/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <TextInput
            type='text'
            name='community-name'
            onChange={this.setState('community-name')}
            inputRef={input => { this.email = input }}
            theme={theme}
            placeholder='Choose a domain name'
          />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.previous}
        showPrevious={false}
        continueText={'Finish Up'}
        />
    </div>
  }
}
