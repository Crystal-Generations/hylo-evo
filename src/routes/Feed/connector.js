import { connect } from 'react-redux'
import { SAMPLE_FEED_ITEMS } from './sampleData'
import { fetchPosts } from './actions'

function mapStateToProps (state) {
  return {
    feedItems: state.posts
  }
}

export const mapDispatchToProps = { fetchPosts }

export default connect(mapStateToProps, mapDispatchToProps)
