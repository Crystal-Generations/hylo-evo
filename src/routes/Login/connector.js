import { connect } from 'react-redux'
import { checkLogin, login, pickError } from './store'

export default connect(pickError, {checkLogin, login})
