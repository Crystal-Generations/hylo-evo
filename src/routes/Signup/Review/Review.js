import React, { Component } from 'react'
import { map } from 'lodash'
import cx from 'classnames'
import LeftSidebar from '../LeftSidebar'
import SignupModalFooter from '../SignupModalFooter'
import UploadImageSection from '../UploadImageSection'
import Pillbox from 'components/Pillbox'

import '../Signup.scss'

export default class AddLocation extends Component {
  constructor () {
    super()
    this.state = {
      edits: {
        name: null,
        email: null,
        password: null,
        location: null
      },
      readOnly: {
        name: true,
        email: true,
        password: true,
        location: true
      }
    }
  }
  handleInputChange = (event, name) => {
    const value = event.target.value
    this.setState({
      edits: {
        ...this.state.edits,
        [name]: value
      }
    })
  }

  makeEditable = (event, name) => {
    this.setState({
      readOnly: {
        ...this.state.edits,
        [name]: false
      }
    })
  }
  submit = () => {
    const location = this.state.location
    this.props.updateUserSettings({location})
    this.props.goToNextStep()
  }

  previous = () => {
    this.props.goToPreviousStep()
  }

  updateSettingDirectly = (key, setChanged) => value => {
    const { edits, changed } = this.state
    // setChanged && setConfirm('You have unsaved changes, are you sure you want to leave?')
    this.setState({
      changed: setChanged ? true : changed,
      edits: {
        ...edits,
        [key]: value
      }
    })
  }

  componentWillMount = () => {
    const { currentUser } = this.props
    if (currentUser && currentUser.settings && currentUser.settings.signupInProgress === 'false') this.props.goBack()
  }

  componentDidMount = () => {
    this.props.fetchMySkills()
  }

  render () {
    const currentAvatarUrl = this.state.edits.avatarUrl

    const { currentUser, uploadImagePending } = this.props
    const skills = currentUser && currentUser.skills && currentUser.skills.toRefArray()
    return <div styleName='wrapper'>
      <LeftSidebar
        header='Everything looking good?'
        body='You can always come back and change your details at any time.'
      />
      <div styleName='detail'>
        <span styleName='white-text step-count'>STEP 4/4</span>
        <br />
        <div styleName='center'>
          {currentUser && <UploadImageSection
            avatarUrl={currentAvatarUrl}
            updateSettingDirectly={this.updateSettingDirectly}
            currentUser={currentUser}
            loading={uploadImagePending}
          />}
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>YOUR NAME</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'name')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.name || currentUser && currentUser.name}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('name')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>YOUR EMAIL</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'email')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.email || currentUser && currentUser.email}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('email')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>LOCATION</div>
          <div styleName='column-center'>
            <input
              styleName='signup-input'
              onChange={(e) => this.handleInputChange(e, 'location')}
              onKeyPress={event => {
                if (event.key === 'Enter') {
                  this.submit()
                  this.props.goToNextStep()
                }
              }}
              autoFocus
              value={this.state.edits.location || currentUser && currentUser.location}
              readOnly={this.state.readOnly.name}
            />
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button' onClick={() => this.makeEditable('location')}>Edit</span>
          </div>
        </div>
        <div styleName='three-column-input'>
          <div styleName='column-left'>SKILLS</div>
          <div styleName='column-center'>
            <div
              styleName={cx('pill-container', {expanded, collapsed: !expanded})}>
              <Pillbox
                pills={map(skills, skill => ({...skill, label: skill.name}))}
                handleInputChange={this.handleInputChange}
                handleAddition={this.handleAddition}
                handleDelete={this.handleDelete}
                editable={false}
                addLabel='Add a Skill'
                placeholder={null}
                suggestions={null}
              />
            </div>
          </div>
          <div styleName='column-right'>
            <span styleName='edit-button'>Edit</span>
          </div>
        </div>
        <div>
          <SignupModalFooter submit={this.submit} previous={this.previous} showPrevious={false} />
        </div>
      </div>
    </div>
  }
}

export function Pill ({skill}) {
  return <span styleName='skill'>
    {skill.name}
  </span>
}
