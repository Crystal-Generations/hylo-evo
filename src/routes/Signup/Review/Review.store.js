export const FETCH_MY_SKILLS = `FETCH_MY_SKILLS`
export const UPDATE_USER_SETTINGS = `UPDATE_USER_SETTINGS`

export function fetchMySkills (limit = 20) {
  return {
    type: FETCH_MY_SKILLS,
    graphql: {
      query: `query ($limit: Int) {
        me {
          id
          skills (first: $limit) {
            items {
              id
              name
            }
          }
        }
      }`,
      variables: { limit }
    },
    meta: {
      extractModel: 'Me'
    }
  }
}
