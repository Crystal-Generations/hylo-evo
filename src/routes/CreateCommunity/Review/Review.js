import React, { Component } from 'react'
import '../CreateCommunity.scss'
import LeftSidebar from '../../Signup/LeftSidebar'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from '../ModalFooter'

const theme = {
  inputStyle: 'modal-input partial',
  wrapperStyle: 'center'
}
export default class Review extends Component {
  setState = () => {

  }
  render () {
    return <div styleName='flex-wrapper'>
      <LeftSidebar
        header='Everything looking good?'
        body='You can always come back and change your details at any time'
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 4/4</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-review'>
          <ReviewTextInput label={'Your Name'} />
          <ReviewTextInput label={'Your Email'} />
          <ReviewTextInput label={'Community Name'} />
          <ReviewTextInput label={'Domain'} />
        </div>
      </div>
      <ModalFooter
        submit={this.submit}
        previous={this.previous}
        showPrevious={false}
        continueText={'Finish Up'}
        />
    </div>
  }
}

export function ReviewTextInput ({label}) {
  return <div styleName='review-input-text-row'>
    <div styleName='review-input-text-label'>
      <span>{label}</span>
    </div>
    <div styleName='review-input-text'>
      <TextInput
        type='text'
        name='community-name'
        value={''}
        theme={theme}
        placeholder="What's the name of your community?"
      />
    </div>
    <div styleName='review-input-edit'>
      <span styleName='edit-button' onClick={() => this.makeEditable('name')}>Edit</span>
    </div>
  </div>
}
