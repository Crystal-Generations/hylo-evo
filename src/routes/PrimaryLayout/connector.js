import { connect } from 'react-redux'
import { SAMPLE_COMMUNITY } from 'routes/Feed/sampleData'

const SAMPLE_USER = {
  id: '1',
  firstName: 'Axolotl',
  lastName: 'Jones',
  avatarUrl: 'https://d3ngex8q79bk55.cloudfront.net/user/13986/avatar/1444260480878_AxolotlPic.png'
}

function mapStateToProps (state) {
  return {
    community: SAMPLE_COMMUNITY,
    currentUser: SAMPLE_USER
  }
}

function mapDispatchToProps (dispatch, props) {
  return {}
}

export default connect(mapStateToProps, mapDispatchToProps)
