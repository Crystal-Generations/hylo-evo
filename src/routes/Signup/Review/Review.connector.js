import { connect } from 'react-redux'
import { push, goBack } from 'react-router-redux'
import getMe from 'store/selectors/getMe'
import { UPLOAD_IMAGE } from 'store/constants'
import { fetchMySkills } from './Review.store'
import { updateUserSettings } from 'store/actions/updateUserSettings'

export function mapStateToProps (state, props) {
  const uploadImagePending = state.pending[UPLOAD_IMAGE]
  return {
    currentUser: getMe(state),
    uploadImagePending
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    updateUserSettings: (changes) => dispatch(updateUserSettings(changes)),
    goToNextStep: () => dispatch(push('/')),
    goToPreviousStep: () => dispatch(push('/signup/add-skills')),
    goBack: () => dispatch(goBack()),
    fetchMySkills: () => dispatch(fetchMySkills())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)
