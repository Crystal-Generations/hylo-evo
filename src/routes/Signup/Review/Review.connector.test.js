import { mapDispatchToProps, mapStateToProps } from './Review.connector'

describe('Review.connector', () => {
  it('should match latest snapshot for mapDispatchToProps', () => {
    expect(mapDispatchToProps).toMatchSnapshot()
  })
  it('should match latest snapshot for mapStateToProps', () => {
    expect(mapStateToProps).toMatchSnapshot()
  })
})
