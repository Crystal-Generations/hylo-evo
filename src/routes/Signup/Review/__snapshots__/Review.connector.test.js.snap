// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review.connector should match latest snapshot for mapDispatchToProps 1`] = `[Function]`;

exports[`Review.connector should match latest snapshot for mapStateToProps 1`] = `[Function]`;
